compiler/src/main/java/com/squareup/anvil/compiler/codegen/ClassReferenceExtensions.kt

```diff
@@ -81,6 +82,10 @@ internal fun ClassReference.checkClassExtendsBoundType(
       classReference = this
     )
 
+  // The boundType is declared explicitly in the annotation. Since all classes extend Any, we can
+  // stop here.
+  if (boundType.fqName == anyFqName) return
```

Comment:
I'm wondering if it would make sense for us to instead modify `allSuperTypeClassReferences()` to include `Any` instead of special casing it here? Efficiency wise this makes sense, but I wonder if we'll potentially run into similar issues later where we're looking for Any and it's not found

Reply:
I don't think so. That might break some existing code, so this is the safer bet. This method also mimics existing APIs from the Kotlin compiler and they exclude `Any` as well.